src/renderers/dom/shared/devtools/ReactDOMUnknownPropertyDevtool.js

```diff
@@ -66,36 +66,56 @@ if (__DEV__) {
       null
     );
 
-    warning(
-      registrationName == null,
-      'Unknown event handler property %s. Did you mean `%s`? %s',
-      name,
-      registrationName,
-      formatSource(source)
-    );
-  };
-
-  var formatSource = function(source) {
-    return source ? `(${source.fileName.replace(/^.*[\\\/]/, '')}:${source.lineNumber})` : '';
+    if (standardName != null) {
+      warning(
+        standardName == null,
+        'Unknown DOM property %s. Did you mean %s? %s',
+        name,
+        standardName,
+        ReactComponentTreeDevtool.getStackAddendumByID(debugID)
+      );
+    } else if (registrationName != null) {
+      warning(
+        registrationName == null,
+        'Unknown event handler property %s. Did you mean `%s`? %s',
+        name,
+        registrationName,
+        ReactComponentTreeDevtool.getStackAddendumByID(debugID)
+      );
+    } else {
+      // We were unable to guess which prop the user intended.
+      // It is likely that the user was just blindly spreading/forwarding props
+      // Components should be careful to only render valid props/attributes.
+      warning(
+        tagName.indexOf('-') >= 0,
```

Comment:
We should at least move this to be in the same place as the `is` check. Ideally, we wouldn't have this duplicated with ReactDOMComponent but if there's no good place for it I guess we can leave this.